streamlit/mmm-explainer/prior_functions.py

```diff
@@ -0,0 +1,111 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+# Imports
+import numpy as np
+import plotly.express as px
+import plotly.graph_objects as go
+import pymc as pm
+from scipy.stats import gaussian_kde
+
+import streamlit as st
+
+
+@st.cache_data  # 👈 Add the caching decorator, make app run faster
+def draw_samples_from_prior(
```

Comment:
I think, we don't need this function. If we need to draw the priors and get an array, we should change to [preliz](https://github.com/arviz-devs/preliz). Documentation [here](https://preliz.readthedocs.io/en/latest/).

We can eliminate this function, and replace in the other files by:
```python
import preliz as pz

def get_distribution(distribution_name, *params):
    # Get the distribution class from preliz
    dist_class = getattr(pz, distribution_name)
    # Create an instance of the distribution with the provided parameters
    return dist_class(*params)

distribution_name = "beta"  # This could be any supported distribution name
alpha, beta = 2, 5  # Parameters for the Beta distribution (Example)

# Create the distribution
dist = get_distribution(distribution_name, alpha, beta)
dist.rvs(n) #n will be the number of values in the array
```

Instead of:
```python
import prior_functions as pf
pf.draw_samples_from_prior(pm.Beta, alpha=alpha, beta=beta, seed=seed)
```

This implementation will help to simplify the UI. Maybe we need only 1 single TAB (Instead of 12) with the input distribution `name` and the parameters for the given distribution. You could even play and create an input dictionary with the params.

Reply:
+1

Reply:
More great feedback, the page is way cleaner now - thanks!